sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java

```diff
@@ -93,24 +94,23 @@ static void configureOtlpExporterBuilder(
       setTimeout.accept(timeout);
     }
 
-    String certificate = config.getString("otel.exporter.otlp." + dataType + ".certificate");
-    if (certificate == null) {
-      certificate = config.getString("otel.exporter.otlp.certificate");
-    }
-    if (certificate != null) {
-      Path path = Paths.get(certificate);
-      if (!Files.exists(path)) {
-        throw new ConfigurationException("Invalid OTLP certificate path: " + path);
-      }
-      byte[] certificateBytes;
-      try {
-        certificateBytes = Files.readAllBytes(path);
-      } catch (IOException e) {
-        throw new ConfigurationException("Error reading OTLP certificate.", e);
-      }
+    byte[] certificateBytes = readFileBytes(config, "otel.exporter.otlp", dataType, "certificate");
+    if (certificateBytes != null) {
       setTrustedCertificates.accept(certificateBytes);
     }
 
+    byte[] clientKeyBytes = readFileBytes(config, "otel.exporter.otlp", dataType, "client.key");
+    byte[] clientKeyChainBytes =
+        readFileBytes(config, "otel.exporter.otlp", dataType, "client.certificate");
+
+    if (clientKeyBytes != null && clientKeyChainBytes == null) {
+      throw new ConfigurationException("Client key provided but certification chain is missing");
+    } else if (clientKeyBytes == null && clientKeyChainBytes != null) {
+      throw new ConfigurationException("Client key chain provided but key is missing");
+    } else {
+      setClientTls.accept(clientKeyBytes, clientKeyChainBytes);
```

Comment:
I believe you're missing a condition. If `clientKeyBytes == null && clientKeyChainBytes == null` we shouldn't call `setClientTls.accept(..)`.

Reply:
since not providing both is not an error I will add condition to only call accept only in positive case when both are provided.  (case when both are not provided will be silently "ignored" as this is expected behaviour)